File: xion-android-demo/app/src/main/java/com/burnt/xiondemo/data/model/OAuthTokens.kt
package com.burnt.xiondemo.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class OAuthTokens(
    @SerialName("access_token") val accessToken: String,
    @SerialName("token_type") val tokenType: String,
    @SerialName("expires_in") val expiresIn: Long,
    @SerialName("refresh_token") val refreshToken: String? = null
)

@Serializable
data class OAuthUserInfo(
    val address: String,
    @SerialName("chain_id") val chainId: String? = null
)

File: xion-ios/XionDemo/Models/OAuthTokens.swift
import Foundation

struct OAuthTokens: Codable {
    let accessToken: String
    let tokenType: String
    let expiresIn: Int64
    let refreshToken: String?

    enum CodingKeys: String, CodingKey {
        case accessToken = "access_token"
        case tokenType = "token_type"
        case expiresIn = "expires_in"
        case refreshToken = "refresh_token"
    }
}

struct OAuthUserInfo: Codable {
    let address: String
    let chainId: String?

    enum CodingKeys: String, CodingKey {
        case address
        case chainId = "chain_id"
    }
}

struct AuthCallbackResult {
    let metaAccountAddress: String
}
